tests/Unit/DependencyInjection/Compiler/HashGeneratorPassTest.php

```diff
@@ -49,11 +49,20 @@ public function testConfigNoContext()
         $config = $this->getBaseConfig();
         $this->extension->load([$config], $container);
         $this->userContextListenerPass->process($container);
-        if ($container->hasDefinition('service_container')) {
-            // symfony 3.3+
-            $this->assertCount(23, $container->getDefinitions());
+        if (method_exists(\ReflectionMethod::class, 'getAttributes')) {
```

Comment:
i would prefer to check the php major version to be more explicit when this is the case.